### PR TITLE
Sparql data connector content type

### LINKED ISSUE #14: Factor out SPARQL data connector
Follow https://docs.plone.org/external/plone.app.dexterity/docs/index.html to create a SPARQL data connector content type that implements the eea.api.dataconnector interface/behaviour as per #12, but is in its own package.

We have the ukstats.dev domain for this, do let's use that in the package name, e.g. ukstats.sparql_dataconnector.

Use `mrbob` as per https://docs.plone.org/external/plone.app.dexterity/docs/prerequisite.html#creating-a-package

### PR DESCRIPTION
**Pull request by nosnickid**
This PR creates a new plone content type called "Sparql Data Connector" that can be used to represent sparql queries.

It allows you to enter a SPARQL endpoint and query, and it supports the EEA `@connector-data` route, which allows it to be used by our chart-builder in volto.

It relies in #138 being merged first; it would have been a merge conflict nightmare otherwise.

Closes #14

### CODE REVIEW

File: plone-5/instance/src/ukstats.sparql_dataconnector/README.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
+.. This README is meant for consumption by humans and pypi. Pypi can render rst files so please do not use Sphinx features.
```
Comment (goofballLogic) [inline plone-5/instance/src/ukstats.sparql_dataconnector/README.rst:1]: README is a bit bogus

File: plone-5/instance/src/ukstats.sparql_dataconnector/LICENSE.GPL [hunk 2]
```diff
@@ -0,0 +1,339 @@
+                    GNU GENERAL PUBLIC LICENSE
```
[the remark this answers is not in the document]
Comment (nosnickid) [reply plone-5/instance/src/ukstats.sparql_dataconnector/LICENSE.GPL:1]: Probably one for @ajtucker - not sure what the usual ONS approach is.

Some of these are just autogenerated from the template we... maybe just deleting the risky stuff is a good idea

File: plone-5/instance/src/ukstats.sparql_dataconnector/README.rst (hunk 1 above)
Comment (nosnickid) [reply plone-5/instance/src/ukstats.sparql_dataconnector/README.rst]: Updated the README to reflect actual extension content

File: plone-5/instance/src/ukstats.sparql_dataconnector/src/ukstats/sparql_dataconnector/adapter.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    """ data provider for connectors """
+
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
+
+    @timing
+    def _get_data(self):
+        """_get_data."""
+        # query = urllib.parse.quote_plus(self.query)
```
[the remark this answers is not in the document]
Comment (nosnickid) [reply plone-5/instance/src/ukstats.sparql_dataconnector/src/ukstats/sparql_dataconnector/adapter.py]: Sorted

File: plone-5/instance/src/ukstats.sparql_dataconnector/src/ukstats/sparql_dataconnector/adapter.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        endpoint_url = self.context.endpoint_url
+        query = self.context.sparql_query
+
+        print(endpoint_url)
```
[the remark this answers is not in the document]
Comment (nosnickid) [reply plone-5/instance/src/ukstats.sparql_dataconnector/src/ukstats/sparql_dataconnector/adapter.py]: This thing started life as a patch to an EEA extension, and getting a debug env working succesfully with plone-5 is something I've still not tackled. But really these should be gone or at least be `logging` calls. I've dropped them for now.

File: volto/ubuntu-install.sh [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+apt-get install -y curl
+apt-get install -y make
+apt-get install -y gcc
+apt-get install -y g++
+
+## Install Node 14
+curl -sL https://deb.nodesource.com/setup_14.x | sudo -E bash -
+sudo apt install nodejs
+
+## Install NVM 16
```
[the remark this answers is not in the document]
Comment (nosnickid) [reply volto/ubuntu-install.sh]: Not really related to this PR, and we kind-of expect devs to have a working node setup. But this doesn't hurt, so I've tweaked the comment. I think callum probably just got node and nvm mixed up.

File: plone-5/instance/src/ukstats.sparql_dataconnector/LICENSE.GPL (hunk 2 above)
[the remark this answers is not in the document]
Comment (nosnickid) [reply plone-5/instance/src/ukstats.sparql_dataconnector/LICENSE.GPL:1]: Seems like GPL2 is fine.

**Review by goofballLogic — APPROVED** [review]
Seems good

**Pull request merged**